server/routers/adRouter.js

```diff
@@ -1,9 +1,30 @@
 const router = require('express').Router();
 const controller = require('../controllers');
 
-router.get('/', controller.admin.get);
-router.post('/', controller.admin.post);
-router.put('/', controller.admin.put);
-router.delete('/', controller.admin.delete);
+router.get('/reservations', controller.reservations.get);
+router.post('/reservations', controller.reservations.post);
+router.put('/reservations', controller.reservations.put);
+router.put('/reservations/:guestId', controller.reservations.put);
+router.delete('/reservations', controller.reservations.delete);
+
+router.get('/rooms', controller.rooms.get);
+router.post('/rooms', controller.rooms.post);
+router.put('/rooms', controller.rooms.put);
+router.put('/rooms/:reservationId', controller.rooms.put);
+router.delete('/rooms', controller.rooms.delete);
+
+router.get('/tasks', controller.tasks.get);
+router.post('/tasks', controller.tasks.post);
+router.put('/tasks', controller.tasks.put);
+router.put('/tasks/:roomId', controller.tasks.put);
```

Comment:
Remove '/tasks/:roomId' put route. The route to '/tasks/:task_id' can handle multiple parameters.